packages/cypress/package.json

```diff
@@ -2,7 +2,7 @@
   "name": "@replayio/cypress",
   "version": "1.7.20",
   "description": "Plugin to record your Cypress tests with Replay",
-  "main": "src/index.js",
+  "main": "dist/src/index.js",
```

Comment:
This is a pretty big change that might affect local development right now. I plan to follow up this PR with some rework in this area to address any pain points that might arise from this.

The main challenge is that we want the IDE to reflect cross-package changes immediately for the fast feedback loop. SImilarly, it would be great if tests could always use the most recent version of all packages. And for various tools to generally understand how those packages are structured to avoid running into corner cases.

The last one is addressed by using dist paths here. The setup becomes clear - we have some set of input files and we also have a set of output files. The published package looks roughly the same as the package looks locally. This also means that I actually still need to take a moment to tweak this setup and publishing step (no more publishing from an alternative directory).

The first problem is fixable using TS project references. They are inconvenient to use since one needs to declare dependencies in them manually (or generate them somehow - which I've done before). It works though.

The second thing (tests) isn't addressed by any of the above. There are 3 popular choices here:
- run a watcher in the background. The user makes a change in an input file, this gets picked up by the watcher and the output gets regenerated. It's fairly straightforward but requires an extra manual step - the user has to actually run this watcher. I think this is fine and it also lends itself to caching etc (it's not a bottleneck in this repo though)
- alias packages - we can force a tool like Jest to resolve `pkg` to just any location. So we can force `@replayio/cypress` to resolve to its `src` directory (over its `dist` that gets picked up by the default). Aliasing like this is tool-specific, it's usually a fairly easy addition to their configs though. 
- spit out some "redirecting" files in dev. I like this strategy as it's quite easy to use - but it's quirky looking and I've only used it through a tool. The tool ([Preconstruct](https://github.com/preconstruct/preconstruct)) is pretty cool and makes package authoring a breeze for me. It's Babel-based though... in a sense that it's using Babel to transpile inputs. So it's flexible but slow (it has never been an issue for me though in the context of monorepos like this one here). This repo already has certain structures and conventions and it might be best - for the time being - to avoid incorporating yet-another-new-tool into this toolchain here.

Reply:
So overall, I think that running an extra watcher would be just good enough here